packages/backend/src/open_payments/receiver/model.test.ts

```diff
@@ -0,0 +1,137 @@
+import { IocContract } from '@adonisjs/fold'
+import { Knex } from 'knex'
+
+import { createTestApp, TestContainer } from '../../tests/app'
+import { Config } from '../../config/app'
+import { initIocContainer } from '../..'
+import { AppServices } from '../../app'
+import { createIncomingPayment } from '../../tests/incomingPayment'
+import { createPaymentPointer } from '../../tests/paymentPointer'
+import { truncateTables } from '../../tests/tableManager'
+import { ConnectionService } from '../connection/service'
+import { Receiver } from './model'
+import { IncomingPaymentState } from '../payment/incoming/model'
+
+describe('Receiver Model', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+  let knex: Knex
+  let connectionService: ConnectionService
+
+  beforeAll(async (): Promise<void> => {
+    deps = initIocContainer(Config)
+    appContainer = await createTestApp(deps)
+    connectionService = await deps.use('connectionService')
+    knex = await deps.use('knex')
+  })
+
+  afterEach(async (): Promise<void> => {
+    jest.useRealTimers()
+    await truncateTables(knex)
+  })
+
+  afterAll(async (): Promise<void> => {
+    await appContainer.shutdown()
+  })
+
+  describe('fromIncomingPayment', () => {
+    test('creates receiver', async () => {
+      const paymentPointer = await createPaymentPointer(deps)
+      const incomingPayment = await createIncomingPayment(deps, {
+        paymentPointerId: paymentPointer.id
+      })
```

Comment:
We should be able to just stub out the instances here instead of actually hitting the services/dbs for example, but that is for a future improvement